extensions/elasticsearch/elasticsearch-7/src/main/java/com/hazelcast/jet/elastic/ElasticSinkBuilder.java

```diff
@@ -0,0 +1,234 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.jet.elastic;
+
+import com.hazelcast.function.ConsumerEx;
+import com.hazelcast.function.FunctionEx;
+import com.hazelcast.function.SupplierEx;
+import com.hazelcast.jet.pipeline.Sink;
+import com.hazelcast.jet.pipeline.SinkBuilder;
+import com.hazelcast.logging.ILogger;
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.ActionRequest;
+import org.elasticsearch.action.DocWriteRequest;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestHighLevelClient;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.io.Serializable;
+
+import static com.hazelcast.jet.impl.util.Util.checkSerializable;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Builder for Elasticsearch Sink
+ *
+ * Requires {@link #clientSupplier(SupplierEx)} and {@link #mapItemFn(FunctionEx)}.
+ *
+ * @param <T>
+ *
+ * @since 4.1
+ */
+public class ElasticSinkBuilder<T> implements Serializable {
+
+    private String name = "elastic";
+    private SupplierEx<? extends RestHighLevelClient> clientSupplier;
+    private ConsumerEx<? super RestHighLevelClient> destroyFn = RestHighLevelClient::close;
+    private SupplierEx<BulkRequest> bulkRequestSupplier = BulkRequest::new;
+    private FunctionEx<? super T, ? extends DocWriteRequest<?>> mapItemFn;
+    private FunctionEx<? super ActionRequest, RequestOptions> optionsFn = (request) -> RequestOptions.DEFAULT;
+    private int preferredLocalParallelism = 2;
+
+    /**
+     * Set the user-friendly source name for this sink
+     *
+     * @param sinkName user-friendly sink name
+     */
+    @Nonnull
+    public ElasticSinkBuilder<T> name(@Nonnull String sinkName) {
```

Comment:
no need for this, sink names can be configured on a per stage basis already

Reply:
Removed.